### PR TITLE
Scaling and range integration

### PR DESCRIPTION
**Pull request by NikolaiShipilov**
* adds support for range and scaling signal parameters for signal generator tool config 
* increases version to 1.2.0

### CODE REVIEW

File: tool/SignalGenerator/signalgenerator.h [hunk 1]
```diff
@@ -39,6 +39,13 @@ namespace daq::streaming_protocol::siggen {
     static const char EXECUTIONTIME[] = "executionTime";
     static const char PROCESSPERIOD[] = "processPeriod";
 
+    static const char RANGE[] = "range";
```
Comment (ml-hbk) [inline tool/SignalGenerator/signalgenerator.h:42]: Could you please use the existing constant from
"include/streaming_protocol/Defines.h"?

File: tool/SignalGenerator/signalgenerator.h [hunk 2]
```diff
@@ -39,6 +39,13 @@ namespace daq::streaming_protocol::siggen {
...
+    static const char LOW[] = "low";
+    static const char HIGH[] = "high";
+    static const char POSTSCALING[] = "postScaling";
+    static const char SCALE[] = "scale";
+    static const char POFFSET[] = "poffset";
```
Comment (ml-hbk) [inline tool/SignalGenerator/signalgenerator.h:47]: Shouldn't this be called "offset". This is also already defined in "Defines.h"

**Review by ml-hbk — APPROVED** [review]

File: tool/SignalGenerator/signalgenerator.h (hunk 1 above)
Comment (NikolaiShipilov) [reply tool/SignalGenerator/signalgenerator.h]: Done

File: tool/SignalGenerator/signalgenerator.h (hunk 2 above)
Comment (NikolaiShipilov) [reply tool/SignalGenerator/signalgenerator.h]: I've replaced local constants with the ones prefixed "META_*" from "Defines.h"

File: CMakeLists.txt [hunk 3]
```diff
@@ -5,7 +5,7 @@ if (NOT CMAKE_MESSAGE_CONTEXT)
     set(CMAKE_MESSAGE_CONTEXT_SHOW ON CACHE BOOL "Show CMake message context")
 endif()
 
-project(streaming_protocol_complete VERSION 1.1.0 LANGUAGES CXX)
+project(streaming_protocol_complete VERSION 1.2.0 LANGUAGES CXX)
```
Comment (nilsRoettgerAtBB) [inline CMakeLists.txt:8]: If we do this we need also a release branch 1.2.0 afterwards if we want to do a release tag. If we want to have just on a working master we should use 1.1.1 for example until we say we release it,

**Review by nilsRoettgerAtBB — APPROVED** [review]

File: CMakeLists.txt (hunk 3 above)
Comment (NikolaiShipilov) [reply CMakeLists.txt:8]: In this case version number follows the rule - "If we adding features / extensions to the document we are increasing y in the x.y.z.". 
I'll create tag and a release branch 1.2.0

**Pull request merged**